app/renderer/src/actions/Inspector.js

```diff
@@ -849,22 +901,76 @@ export function setAwaitingMjpegStream(isAwaiting) {
   };
 }
 
+export function setGestureUploadErrors(errors) {
+  return (dispatch) => {
+    dispatch({type: SET_GESTURE_UPLOAD_ERROR, errors});
+  };
+}
+
+export function uploadGesturesFromFile(fileList) {
+  return async (dispatch) => {
+    const gestures = await readTextFromUploadedFiles(fileList);
+    const invalidGestures = {};
+    const parsedGestures = [];
+    gestures.forEach((gestureFilePromise) => {
+      const {fileName, content, error} = gestureFilePromise;
```

Comment:
if it a promise then should it be awaited?

Reply:
or maybe the arg name is confusing

Reply:
Addressed this in latest patch.